### ISSUE TITLE
🐛 BUG: "\" in MD Setup: breaks build and dev

### ISSUE THREAD

**Issue by tusamni** [withastro/astro#3230]
### What version of `astro` are you using?

1.0.0-beta.19

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Windows 11

### Describe the Bug

I discovered a frustrating compiler issue.

In a random .MD file, I had an accidental "\" at the end of my layout import. This backslash causes dev to break and build hangs at `[build] Collecting build information...`

Here's the code:

```
setup: |
  import Layout from "@/layouts/PostLayout"\
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-c3xupu?file=src%2Fpages%2Fposts%2Findex.md&on=stackblitz

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by natemoo-re** [comment]
Yep, sorry for the frustration! This is definitely a compiler issue, I'm going to move this over to the compiler repo.
